osbuild/util/pe32p.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/python3
+"""
+Utility functions to inspect PE32+ files
+
+
+"""
+
+import enum
+import io
+import os
+import struct
+import sys
+from collections import namedtuple
+from typing import Iterator, List, Optional, Union
+
+PathLike = Union[str, bytes, os.PathLike]
+
+CoffFormat = "4sHHIIIHH"
+CoffHeader = namedtuple(
+    "CoffHeader",
+    "Signature "
+    "Machine "
+    "NumberOfSections "
+    "TimeDateStamp "
+    "PointerToSymbolTable "
+    "NumberOfSymbols "
+    "SizeOfOptionalHeader "
+    "Characteristics"
+)
+
+
+SectionFormat = "8sIIIIIIHHI"
+SectionHeader = namedtuple(
+    "SectionHeader",
+    "Name "
+    "VirtualSize "
+    "VirtualAddress "
+    "SizeOfRawData "
+    "PointerToRawData "
+    "PointerToRelocations "
+    "PointerToLinenumbers "
+    "NumberOfRelocations "
+    "NumberOfLinenumbers "
+    "Characteristics "
+)
+
+
+class SectionFlags(enum.Flag):
+    ALIGN_1BYTES = 0x00100000
+    ALIGN_2BYTES = 0x00200000
+    ALIGN_4BYTES = 0x00300000
+    ALIGN_8BYTES = 0x00400000
+    ALIGN_16BYTES = 0x00500000
+    ALIGN_32BYTES = 0x00600000
+    ALIGN_64BYTES = 0x00700000
+    ALIGN_128BYTES = 0x00800000
+    ALIGN_256BYTES = 0x00900000
+    ALIGN_512BYTES = 0x00A00000
+    ALIGN_1024BYTES = 0x00B00000
+    ALIGN_2048BYTES = 0x00C00000
+    ALIGN_4096BYTES = 0x00D00000
+    ALIGN_8192BYTES = 0x00E00000
+    ALIGN_MASK = 0x00F00000
+    ALIGN_DEFAULT = ALIGN_16BYTES
+
+
+OptionalFormat = "HBBIIIIIQIIHHHHHHIIIIHHQQQQII"
+OptionalHeader = namedtuple(
+    "OptionalHeader",
+    # Standard fields
+    "Magic "
+    "MajorLinkerVersion "
+    "MinorLinkerVersion "
+    "SizeOfCode "
+    "SizeOfInitializedData "
+    "SizeOfUninitializedData "
+    "AddressOfEntryPoint "
+    "BaseOfCode "
+    # Windows-Specific fields (PE32+)
+    "ImageBase "
+    "SectionAlignment "
+    "FileAlignment "
+    "MajorOperatingSystemVersion "
+    "MinorOperatingSystemVersion "
+    "MajorImageVersion "
+    "MinorImageVersion "
+    "MajorSubsystemVersion "
+    "MinorSubsystemVersion "
+    "Reserved1 "
+    "SizeOfImage "
+    "SizeOfHeaders "
+    "CheckSum "
+    "Subsystem "
+    "DllCharacteristics "
+    "SizeOfStackReserve "
+    "SizeOfStackCommit "
+    "SizeOfHeapReserve "
+    "SizeOfHeapCommit "
+    "LoaderFlags "
+    "NumberOfRvaAndSizes "
+)
+
+
+def seek_to_coff(f, check: bool = False):
```

Comment:
Maybe prefix private functions with an underscore? I assume `seek_to_coff()` (and some of the other helpers) are not meant to be exported?

The current module makes it hard to really see which is the public API, and which is not.

Reply:
Fair point, let me change that.